operationsgateway_api/src/routes/filters.py

```diff
@@ -0,0 +1,176 @@
+
+
+@router.get(
+    "/users/filters",
+    summary="Get a list of the user's favourite filters",
+    response_description="List of favourite filters",
+    tags=["Filters"],
+)
+@endpoint_error_handling
+async def get_favourite_filters(access_token: AuthoriseToken):
+    """
+    Get a list of the user's favourite filters that belong to the user sending the
+    request
+    """
+
+    token_payload = JwtHandler.get_payload(access_token)
+    username = token_payload["username"]
+    log.info("Getting favourite filters for user %s", username)
+
+    return await FavouriteFilter.get_list(username)
+
+
+@router.get(
+    "/users/filters/{_id}",
+    summary="Get a list of the user's favourite filters",
+    response_description="List of favourite filters",
+    tags=["Filters"],
+)
+@endpoint_error_handling
+async def get_favourite_filter_by_id(
+    _id: Annotated[
+        str,
+        Path(
+            ...,
+            description="`_id` of the filter to remove from the database",
+        ),
+    ],
+    access_token: AuthoriseToken,
+):
+    """
+    Get a specific favourite filter given its ID
+    """
+
+    token_payload = JwtHandler.get_payload(access_token)
+    username = token_payload["username"]
+    log.info("Getting favourite filter of _id '%s' for user '%s'", _id, username)
+    return await FavouriteFilter.get_by_id(username, _id)
+
+
+@router.post(
+    "/users/filters",
+    summary="Store a favourite filter into the user's document in the database",
+    response_description="List of favourite filters",
+    tags=["Filters"],
+)
+@endpoint_error_handling
+async def create_favourite_filter(
+    access_token: AuthoriseToken,
+    name: str = Query(
+        "",
+        description="Name of the favourite filter",
+    ),
+    filter: str = Query(  # noqa: A002
+        "",
+        description="Contents of the favourite filter",
+    ),
+):
+    """
+    Create a favourite filter for the user sending the request and save it in the user's
+    document in the database
+    """
+
+    token_payload = JwtHandler.get_payload(access_token)
+    username = token_payload["username"]
+
+    log.info("Creating favourite filter for user '%s'", username)
+    new_filter = FavouriteFilter(username, name, filter)
+    filter_id = await new_filter.create()
+
+    return JSONResponse(
+        str(filter_id),
+        status_code=status.HTTP_201_CREATED,
+        headers={"Location": f"/users/filters/{filter_id}"},
+    )
+
+
+@router.patch(
+    "/users/filters/{_id}",
+    summary="Update a user's favourite filter",
+    response_description="Confirmation that the filter has been successfully updated",
+    tags=["Filters"],
+)
+@endpoint_error_handling
+async def update_favourite_filter(
+    access_token: AuthoriseToken,
+    _id: Annotated[
+        str,
+        Path(
+            ...,
+            description="`_id` of the favourite filter to update",
+        ),
+    ],
+    name: str = Query(
+        "",
```

Comment:
Again I'm not sure of the implications of this defaulting to `""`. How can a user change only the `name`, for example? If they omit the `value` it would default to `""` and so overwrite the value they wanted to persist with an empty string?

Reply:
I've refactored this endpoint so you can partially update a filter and when nothing is updated, nothing is done. See 4c3feec77ac8e923e65a991ad1deb888e98caab7